library/Zend/Di/InstanceManager.php

```diff
@@ -502,7 +502,12 @@ protected function createHashForValues($classOrAlias, $paramValues)
                     $hashValue .= $param . '|';
                     break;
                 case 'array':
-                    $hashValue .= 'Array|';
+                    try {
+                        $hashValue .= serialize($param) . '|';
+                    } catch (\Exception $ex) {
```

Comment:
@noopable `serialize()` doesn't raise an exception, it raises a warning, which you cannot catch via try/catch. `Zend\Stdlib\ErrorHandler` provides a mechanism for shimming in a context-specific error handler from which you can then recover or raise warnings/exceptions/etc. Just wrap the call with:

- [ErrorHandler::start](https://github.com/zendframework/zf2/blob/master/library/Zend/Stdlib/ErrorHandler.php#L52)
- [ErrorHandler::stop](https://github.com/zendframework/zf2/blob/master/library/Zend/Stdlib/ErrorHandler.php#L68)